homeassistant/components/modbus/binary_sensor.py

```diff
@@ -84,14 +84,13 @@ async def async_setup_platform(
             CONF_NAME: "no name",
             CONF_BINARY_SENSORS: config[CONF_INPUTS],
         }
-        config = None
 
     for entry in discovery_info[CONF_BINARY_SENSORS]:
-        if CONF_HUB in entry:
-            # from old config!
-            hub: ModbusHub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
-        else:
-            hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        hub: ModbusHub = (
```

Comment:
I say thanks for the review comment received and the interesting solutions, I searched for a simple solution, and I found one.

Reply:
I say that is way less readable than cast. But i suppose that is a matter of opinion.

I would have done:
```
data = hass.data[MODBUS_DOMAIN]
if discovery_info:
  hub_name = discovery_info[CONF_HUB]]
else:
  hub_name = entry[CONF_NAME]
hub = cast(Modbus, data[hub_name])
```

Reply:
yes that is an alternative, but then you would have had the same problem with hub_name being without type....so that is really the same as I started out with.

Apart from that, it is better to write "hub: ModbusHub = data[hub_name]

I started with 
```
if .... 
   hub = ...
else
   hub = ...
```
This is still my preferred version, but as you noted, that declares hub without type, is that really worse than declaring hub_name without type ?